File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F15.P2.java
class Main {
    // BEGIN PROBE F15.P2
    static <T extends Comparable<T>> T maxOf(T a, T b) {
        return a.compareTo(b) > 0 ? a : b;
    }

    static int probe() {
        int m = maxOf(3, 5);
        return m;
    }
    // END PROBE F15.P2

    public static void main(String[] args) {
        System.out.println(probe());
    }
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F15.P2.cpp
#include <concepts>
#include <cstdint>
#include <cstdio>

std::int32_t biggest() {
// BEGIN PROBE F15.P2
    auto max_of = []<std::totally_ordered T>(T x, T y) { return x < y ? y : x; };
    std::int32_t m = max_of(3, 5);
    return m;
// END PROBE F15.P2
}

int main() {
    std::printf("%d\n", biggest());
}
